frontend/app/routes/public/apply/$id/child/email.tsx

```diff
@@ -81,13 +86,50 @@ export async function action({ context: { appContainer, session }, params, reque
     return data({ errors: transformFlattenedError(parsedDataResult.error.flatten()) }, { status: 400 });
   }
 
-  saveApplyState({ params, session, state: { email: parsedDataResult.data.email } });
+  const isNewEmail = state.email !== parsedDataResult.data.email;
+  const verificationCode = isNewEmail || state.verifyEmail === undefined ? verificationCodeService.createVerificationCode('anonymous') : state.verifyEmail.verificationCode;
+
+  invariant(state.communicationPreferences, 'Expected state.communicationPreferences to be defined');
+  if (isNewEmail) {
+    const preferredLanguageService = appContainer.get(TYPES.domain.services.PreferredLanguageService);
+    const preferredLanguage = preferredLanguageService.getLocalizedPreferredLanguageById(state.communicationPreferences.preferredLanguage, locale).name;
+
+    await verificationCodeService.sendVerificationCodeEmail({
+      email: parsedDataResult.data.email,
+      verificationCode,
+      preferredLanguage: preferredLanguage === PREFERRED_LANGUAGE.en ? 'en' : 'fr',
+      userId: 'anonymous',
+    });
+  }
+
+  saveApplyState({
+    params,
+    session,
+    state: {
+      email: parsedDataResult.data.email,
+      emailVerified: isNewEmail ? false : state.emailVerified,
+      ...(isNewEmail && {
+        verifyEmail: {
+          verificationCode,
+          verificationAttempts: 0,
+        },
+      }),
+    },
+  });
 
   if (state.editMode) {
+    // Redirect to /verify-email only if emailVerified is false
+    if (isNewEmail || !state.emailVerified) {
+      return redirect(getPathById('public/apply/$id/child/verify-email', params));
+    }
     return redirect(getPathById('public/apply/$id/child/review-adult-information', params));
   }
 
-  return redirect(getPathById('public/apply/$id/child/verify-email', params));
+  if (isNewEmail || !state.emailVerified) {
+    return redirect(getPathById('public/apply/$id/child/verify-email', params));
+  }
+
+  return redirect(getPathById('public/apply/$id/child/dental-insurance', params));
```

Comment:
Correct me if I'm wrong:

```suggestion
  return redirect(getPathById('public/apply/$id/child/review-child-information', params));
```